discovery-kubernetes-api/src/test/scala/akka/discovery/kubernetes/KubernetesApiServiceDiscoverySpec.scala

```diff
@@ -45,6 +45,44 @@ class KubernetesApiServiceDiscoverySpec extends WordSpec with Matchers {
         "cluster.local") shouldBe List.empty
     }
 
+    "return a single result per host with no port when no port name is requested" in {
```

Comment:
Great to add this test. Might be good to document in a little more detail *why* this behavior is desired. Perhaps even explicitly link to https://github.com/jroper/akka-grpc-ddata-shopping-cart/tree/istio-future and explain it is needed for that use case?

Reply:
Document where? In the documentation? Or in comments?

I actually think this is a bug fix, regardless of why it's being implemented. If you do a lookup on any arbitrary discovery implementation, and don't specify any port in the lookup, do you expect to get a port back in the resolved target? I would have thought no, you certainly won't get one back from the dns implementation, it only does a SRV lookup if you specify a port. So the same should go for the kubernetes-api implementation, I just expect it to return me a list of hosts without ports if I don't specify a port to lookup. So I'm not sure that any special explanation is needed, it's now behaving intuitively, if not according to spec, whereas before it was behaving unintuitively, and that was probably a bug.

I think once the Istio patch is merged, we should add some documentation (perhaps to the OpenShift guide) on how to form an Akka cluster in Istio, and the need to not specify a port can be mentioned there. Otherwise I don't think there's any need to document it.

Reply:
> Document where? In the documentation? Or in comments?

I was thinking of in a comment along with this test

> If you do a lookup on any arbitrary discovery implementation, and don't specify any port in the lookup, do you expect to get a port back in the resolved target

In some other context I might desire to get back *all* the ports. After all, devils' advocate, what good is getting back a service IP if I still don't receive enough information to actually connect to it? Especially if the information is readily available from the discovery backend? If someone asks for that feature in the future, this test will fail. I would like to explain to future-me that this test doesn't just describe a malleable status-quo, but there are actually use cases depending on this behavior.

What's 'intuitive' can change a lot based on context ;).

Reply:
Fair points. I've added the comment.